tgui/packages/tgui/interfaces/BankingComputer/CrewAccounts.tsx

```diff
@@ -0,0 +1,224 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author mloccy (https://github.com/mloccy)
+ * @license MIT
+ */
+
+import { useBackend, useLocalState } from "../../backend";
+import { Button, Divider, Dropdown, Input, Section, Stack, Table } from "../../components";
+import { SortDirection } from "../PlayerPanel/constant";
+import { Header } from "../PlayerPanel/Header";
+import { TransferButton } from "./TransferButton";
+import { ColumnSortField, CREDIT_SIGN, CrewAccount, CrewColumnSortConfig, SearchFilter } from "./type";
+
+interface CrewAccountLineProps {
+  account: CrewAccount
+}
+
+const CrewAccountLine = (props: CrewAccountLineProps, context) => {
+  const { data, act } = useBackend(context);
+  return (
+    <>
+      <Table.Cell className="crewAccountCell" bold>{props.account.name}</Table.Cell>
+      <Table.Cell className="crewAccountCell">{props.account.job}</Table.Cell>
+      <Table.Cell className="crewAccountCell"><Button title={"Edit Salary"} onClick={() => act("edit_wage", { "id": props.account.id })} icon="pen" /></Table.Cell>
+      <Table.Cell className="crewAccountCell">{props.account.wage.toLocaleString()}{CREDIT_SIGN}</Table.Cell>
+      <Table.Cell className="crewAccountCell">{props.account.balance.toLocaleString()}{CREDIT_SIGN}</Table.Cell>
+      <Table.Cell className="crewAccountCell"><TransferButton frozen={props.account.frozen} id={props.account.id} type={"crew"} /></Table.Cell>
+    </>
+  );
+};
+
+
+interface CrewAccountsTableHeaderProps {
+  onSortClick: (config: CrewColumnSortConfig) => any,
+  currentColumnConfig: CrewColumnSortConfig
+}
+const CrewAccountsTableHeader = (props: CrewAccountsTableHeaderProps, context) => {
+  const setColumnConfig = (field: ColumnSortField) => {
+    if (props.currentColumnConfig === null) {
+      let config = { "field": field, dir: SortDirection.Asc };
+      props.onSortClick(config);
+    } else {
+      let config = props.currentColumnConfig;
+      if (config.dir === SortDirection.Asc) {
+        config.dir = SortDirection.Desc;
+      } else {
+        config.dir = SortDirection.Asc;
+      }
+
+      config.field = field;
+      props.onSortClick(config);
```

Comment:
This mutates the current `config` that's passed in via a prop. Rewrite to not do so.